### DISCUSSION TITLE
Rich-text editor in pins

### DISCUSSION THREAD

**Discussion by daun** [jacksleight/statamic-bard-texstyle#68] [General]
I'm currently trying out the new Pins feature for footnotes. It's a fantastic addition, really impressed by how easy it is to set up.

Now I'm wondering if it's feasible to implement a (very) simple rich-text editor for the pin contents. I've seen TipTap demos where the footnotes are nested TipTap editors themselves, but I can't judge the complexity here. My workaround for now is Markdown, which is fine for bold/italic but missing some convenient features around linking to existing entries and assets.

**Comment by jacksleight** [comment]
You can use a bard field in a pin, so that should give you all the options you need. The only limitation is you can't use sets or pins inside those nested bard fields.

**Reply by daun** [reply]
Oh right! I missed that part in the docs, I figured it's text or textarea only. Sorry for being dense :)

**Comment by daun** [comment]
Posting my solution here in case anyone needs something similar.

This will create a very minimal Bard field for rich-text footnotes allowing bold, italic, and links.

```php
// config/statamic/bard_texstyle.php

/*
|--------------------------------------------------------------------------
| Pins
|--------------------------------------------------------------------------
|
*/

'pins' => [

    'footnote' => [
        'display' => 'Footnote',
        'icon' => 'symbol',
        'ident' => '*',
        'instructions' => 'A footnote rendered below the text',
        'fields' => [
            'content' => [
                'type' => 'bard',
                'display' => 'Text',
                'hide_display' => true,
                'allow_source' => false,
                'fullscreen' => false,
                'buttons' => [
                    'bold',
                    'italic',
                    'anchor',
                ],
            ],
        ],
    ],

],
```
